modules/vault-cluster/main.tf

```diff
@@ -213,3 +214,27 @@ data "aws_iam_policy_document" "vault_s3" {
     ]
   }
 }
+
+resource "aws_iam_role_policy" "vault_aws_EC2_IAM_Auth" {
+  count  = "${var.enable_EC2_IAM_Auth ? 1 : 0}"
```

Comment:
If `enable_EC2_IAM_Auth` is a boolean, the `? 1 : 0` bit is redundant, as a `true` is treated as a `1` and a false as a `0`.